### ISSUE TITLE
Error: Can't use $id with Array.

### ISSUE THREAD

**Issue by brooth** [Automattic/mongoose#8562]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
The query which works fine in mongodb fails in mongoose.

**If the current behavior is a bug, please provide the steps to reproduce.**
I have a list of DBRefs
```
        members: {
            $ref: string;
            $id: Types.ObjectId;
            $db: string;
        }[];
```
I am able to find all items that contain at least one member using this query:
```
db.chats.findOne({ members: {$elemMatch: {$ref: 'canals', $id: ObjectId('5e39af4a7f71bcc0a1a10392')}}});
```
, but the same query fails if I try to run it via mongoose:
```
 await Chat.findOne({
      members: { $elemMatch: { '$id': main_canal._id, $ref: 'canals' } },
});
```
, the error:

```
  Can't use $id with Array.
    at DocumentArrayPath.SchemaArray.castForQuery (***/node_modules/mongoose/lib/schema/array.js:359:13)
    at DocumentArrayPath.cast$elemMatch (***/node_modules/mongoose/lib/schema/array.js:442:23)
    at DocumentArrayPath.SchemaArray.castForQuery (***/node_modules/mongoose/lib/schema/array.js:362:19)
    at DocumentArrayPath.SchemaType.castForQueryWrapper (***/node_modules/mongoose/lib/schematype.js:1353:17)
    at cast (***/node_modules/mongoose/lib/cast.js:296:39)
    at model.Query.Query.cast (***/node_modules/mongoose/lib/query.js:4683:12)
    at castQuery (***/node_modules/mongoose/lib/query.js:4495:18)
    at model.Query.Query._findAndModify (***/node_modules/mongoose/lib/query.js:3439:23)
    at model.Query.<anonymous> (***/node_modules/mongoose/lib/query.js:3021:8)
    at model.Query._wrappedThunk [as _findOneAndUpdate] (***/node_modules/mongoose/lib/helpers/query/wrapThunk.js:16:8)
    at process.nextTick (***/node_modules/kareem/index.js:369:33)
    at process._tickCallback (internal/process/next_tick.js:61:11)
```

**What is the expected behavior?**
I'd like it doesn't fail.

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Node.js - v10.15.0
Mongoose - 5.8.11
MongoDB - 4.0.9

**Comment by achillesoo7** [comment]
Hey,
According to your given query and error, there can be two issues:
One is, you have to cast your normal _id to actual ObjectId with Schema.Types.ObjectId(_id)

Or You might be trying to query an array, which can be done with $unwind and $match aggregate operators.
https://github.com/Automattic/mongoose/issues/3090

For being more specific, can you please provide an idea about the kind of collection, you are trying to query or any sort of example.

**Comment by brooth** [comment]
Fixed here 2c6546d. Closing
